File: cursoqualidadecodigojava/src/main/java/com/ifood/cursoqualidadecodigo/Estacionamento.java
package com.ifood.cursoqualidadecodigo;

import com.ifood.cursoqualidadecodigo.model.Carro;

import java.util.ArrayList;
import java.util.List;

public class Estacionamento {

    private final List<Carro> carros;

    public Estacionamento() {
        carros = new ArrayList<Carro>();
    }

    public void incluirCarro(Carro carro) {
        carros.add(carro);
    }

    public void removerCarro(Carro carro) {
        carros.remove(carro);
    }

    public int contarCarros() {
        return carros.size();
    }

}

File: cursoqualidadecodigonode/src/estacionamento.js

function Estacionamento() {
    this.__carros = [];
    this.incluirCarro = function(carro) {
        this.__carros.push(carro);
    };
    this.removerCarro = function(carro) {
        for (let i = 0; i < this.__carros.length; i++) {
            if (this.__carros[i].placa === carro.placa) {
                this.__carros.splice(i, 1);
            }
        }
    };
    this.contarCarros = function() {
        return this.__carros.length;
    }
}
module.exports = Estacionamento;